File: GridPatternExample/src/main/scala/org/akka/essentials/grid/controller/WorkSchedulerActor.scala
package org.akka.essentials.grid.controller
import akka.actor.Actor
import akka.actor.ActorLogging
import akka.util.duration._

class WorkSchedulerActor extends Actor with ActorLogging {

  def receive: Receive = {
    case value:Int =>
      val cancellable = context.system.scheduler.scheduleOnce(value.milliseconds, sender,  SendWork)
    case _ =>
      log.info("Received work sending request")
      val cancellable = context.system.scheduler.scheduleOnce(6000 milliseconds, sender,  SendWork)
  }
}

File: GridPatternExample/src/main/java/org/akka/essentials/grid/controller/WorkSchedulerActor.java
package org.akka.essentials.grid.controller;

import java.util.concurrent.TimeUnit;

import scala.concurrent.duration.Duration;
import akka.actor.UntypedActor;
import akka.event.Logging;
import akka.event.LoggingAdapter;

public class WorkSchedulerActor extends UntypedActor {

	LoggingAdapter log = Logging.getLogger(getContext().system(), this);

	@Override
	public void onReceive(Object message) throws Exception {

		if (message instanceof Integer) {
			getContext()
					.system()
					.scheduler()
					.scheduleOnce(
							Duration.create(((Integer) message).intValue(),
									TimeUnit.SECONDS), getContext().sender(),
							"SendWork",getContext().dispatcher());
		} else if (message instanceof String) {
			log.info("Recieved work sending request");
			getContext()
					.system()
					.scheduler()
					.scheduleOnce(Duration.create(1000, TimeUnit.MILLISECONDS),
							getContext().sender(), "SendWork",getContext().dispatcher());
		}
	}

}
